### PR TITLE
[exporter/debug] format log records as one-liners in `normal` verbosity

### PR DESCRIPTION
**Pull request by andrzej-stencel**
#### Description

Changes the behavior of `normal` verbosity of the Debug exporter for logs to display each log record in one line of text.

Note that if the body of the log record contains newlines, the output will be displayed in multiple lines.

This pull request is part of https://github.com/open-telemetry/opentelemetry-collector/issues/7806; it implements the change for logs. The changes for metrics and [traces](https://github.com/open-telemetry/opentelemetry-collector/pull/10280) will be proposed in separate pull requests.

The implementation in this pull request does not display any details on the resource or the scope of the logs. I would like to propose displaying the resource and the scope as separate lines in a separate pull request.

This change applies to the Debug exporter only. The behavior of the Logging exporter remains unchanged. To use this behavior, switch from the deprecated Logging exporter to Debug exporter.

#### Link to tracking issue

- https://github.com/open-telemetry/opentelemetry-collector/issues/7806

#### Testing

Added unit tests for the formatter.

#### Documentation

Described the formatting in the Debug exporter's README.

### CODE REVIEW

**Comment by andrzej-stencel** [comment]
Test coverage should be improved by https://github.com/open-telemetry/opentelemetry-collector/pull/10242.

File: exporter/debugexporter/README.md [hunk 1]
```diff
@@ -60,8 +60,22 @@ Here's an example output:
 
 ### Normal verbosity
 
-With `verbosity: normal`, the exporter's behavior is currently the same as with `verbosity: basic`.
-See above for more details.
+With `verbosity: normal`, the exporter outputs about one line for each telemetry record.
```
Comment (rogercoll) [inline exporter/debugexporter/README.md:63]: Should we document which LogRecord fields will be included into the log line?

```suggestion
With `verbosity: normal`, the exporter outputs about one line for each telemetry record, including its body and attributes.
```

File: exporter/debugexporter/internal/normal/logs.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	var buffer bytes.Buffer
+	for i := 0; i < ld.ResourceLogs().Len(); i++ {
+		resourceLog := ld.ResourceLogs().At(i)
+		for j := 0; j < resourceLog.ScopeLogs().Len(); j++ {
+			scopeLog := resourceLog.ScopeLogs().At(j)
+			for k := 0; k < scopeLog.LogRecords().Len(); k++ {
+				logRecord := scopeLog.LogRecords().At(k)
+				logAttributes := writeAttributes(logRecord.Attributes())
+
+				logString := fmt.Sprintf("%s %s", logRecord.Body().AsString(), strings.Join(logAttributes, " "))
```
Comment (rogercoll) [inline exporter/debugexporter/internal/normal/logs.go:35]: Would it make sense to include the severity text value? (log level) I think this value gives important context about the body message

**Review by rogercoll — COMMENTED** [review]
LGTM, just a few minor comments.

File: exporter/debugexporter/internal/normal/logs.go (hunk 2 above)
Comment (andrzej-stencel) [reply exporter/debugexporter/internal/normal/logs.go:35]: Good point. Can we make an enhancement in a separate PR? There might be other suggestions to improve/change the output and I think it'll be easier to discuss each one separately.

File: exporter/debugexporter/exporter.go [hunk 3]
```diff
@@ -28,10 +29,16 @@ type debugExporter struct {
 }
 
 func newDebugExporter(logger *zap.Logger, verbosity configtelemetry.Level) *debugExporter {
+	var logsMarshaler plog.Marshaler
+	if verbosity == configtelemetry.LevelDetailed {
+		logsMarshaler = otlptext.NewTextLogsMarshaler()
+	} else {
+		logsMarshaler = normal.NewNormalLogsMarshaler()
+	}
```
[the remark this answers is not in the document]
Comment (andrzej-stencel) [reply exporter/debugexporter/exporter.go:37]: Hm yes, that is an option. What would be the benefits of this?

File: exporter/debugexporter/README.md (hunk 1 above)
Comment (andrzej-stencel) [reply exporter/debugexporter/README.md:63]: Yes, this is probably useful. :+1:

File: exporter/debugexporter/internal/normal/logs.go (hunk 2 above)
Comment (rogercoll) [reply exporter/debugexporter/internal/normal/logs.go:35]: Sounds good, let's bring the discussion in another PR/issue

File: exporter/debugexporter/exporter.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (rogercoll) [reply exporter/debugexporter/exporter.go:37]: The main benefit would be the encapsulation of the "text" marshaler in a single struct/object. Also, a single test file would be needed to their similar functionality. But as mentioned, I don't have a strong opinion on that, we can definitely go with the normal marshaler entity for this PR and reconsider it in the future if we keep adding text marshalers :+1:

[the remark this answers is not in the document]
Comment (andrzej-stencel) [reply exporter/debugexporter/exporter.go:37]: Sure. I agree that consolidating the unit tests would be very useful. I'm not entirely sure as to keeping two effectively different implementations of marshaling in one place, but I'll keep this in mind when further working on this.

File: .chloggen/debug-exporter-normal-verbosity-logs.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```
Comment (mx-psi) [inline .chloggen/debug-exporter-normal-verbosity-logs.yaml:4]:
```suggestion
change_type: enhancement
```

Same as https://github.com/open-telemetry/opentelemetry-collector/pull/10280#discussion_r1624076976, this is not breaking

**Review by mx-psi — APPROVED** [review]

**Pull request merged**